src/algebra/periodic.lean

```diff
@@ -0,0 +1,264 @@
+/-
+Copyright (c) 2021 Benjamin Davidson. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Benjamin Davidson
+-/
+import data.int.parity
+
+/-!
+# Periodicity
+
+In this file we define and then prove facts about periodic and antiperiodic functions.
+
+## Main definitions
+
+* `function.periodic`: A function `f` is *periodic* if `∀ x, f (x + c) = f x`.
```

Comment:
Floris and I discussed doing it that way and then using `to_additive` but we agreed it wasn't worthwhile since we couldn't think of any instance where the multiplicative version would be useful.